lib/grape/cookies.rb

```diff
@@ -32,7 +32,7 @@ def each(&block)
     end
 
     def delete(name, opts = {})
-      options = opts.merge(value: 'deleted', expires: Time.at(0))
+      options = opts.merge!(value: 'deleted', expires: Time.at(0))
```

Comment:
When you say local, local to the class? Clearly here `opts` is not a local variable. 

```ruby
2.2.4 :001 > def doit(opts)
2.2.4 :002?>     opts[:foobar] = 1
2.2.4 :003?>   end
 => :doit 
2.2.4 :004 > 
2.2.4 :005 >   opts = {}
 => {} 
2.2.4 :006 > doit(opts)
 => 1 
2.2.4 :007 > p opts
{:foobar=>1}
 => {:foobar=>1} 
```

Modifying a hash within the function causes a side effect of modifying the caller's hash. So it's bad practice, finding bugs caused by such side effect is really hard.